common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -325,6 +330,15 @@ private void processPublish(ChannelHandlerContext ctx, MqttPublishMessage mqttMs
                 handleGatewayPublishMsg(ctx, topicName, msgId, mqttMsg);
                 transportService.reportActivity(deviceSessionCtx.getSessionInfo());
             }
+        } else if (sparkPlugSessionHandler != null) {
+            try {
```

Comment:
See how we implement similar logic for the gateway. 
```
            if (gatewaySessionHandler != null) {
                handleGatewayPublishMsg(ctx, topicName, msgId, mqttMsg);
                transportService.reportActivity(deviceSessionCtx.getSessionInfo());
            }
```
You should have similar structure and add all the logic to the `handleSparkplugPublishMsg`

Reply:
except `transportService.reportActivity(deviceSessionCtx.getSessionInfo());`

Reply:
Ok